src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponentSpec.scala

```diff
@@ -100,4 +100,64 @@ class ClusterComponentSpec extends TestComponent with FlatSpecLike {
     assert(c2status.destroyedDate.nonEmpty)
     c2status.hostIp shouldBe None
   }
+
+  it should "get by labels" in isolatedDbTest {
+    val c1 = Cluster(
+      clusterName = "name1",
+      googleId = UUID.randomUUID(),
+      googleProject = "dsp-leo-test",
+      googleServiceAccount = "not-a-service-acct@google.com",
+      googleBucket = "bucket1",
+      clusterUrl = Cluster.getClusterUrl("dsp-leo-test", "name1"),
+      operationName = "op1",
+      status = ClusterStatus.Unknown,
+      hostIp = Some("numbers.and.dots"),
+      createdDate = Instant.now(),
+      destroyedDate = None,
+      labels = Map("bam" -> "yes", "vcf" -> "no", "foo" -> "bar"),
+      jupyterExtensionUri = None)
+
+    val c2 = Cluster(
+      clusterName = "name2",
+      googleId = UUID.randomUUID(),
+      googleProject = "dsp-leo-test",
+      googleServiceAccount = "not-a-service-acct@google.com",
+      googleBucket = "bucket2",
+      clusterUrl = Cluster.getClusterUrl("dsp-leo-test", "name2"),
+      operationName = "op2",
+      status = ClusterStatus.Running,
+      hostIp = None,
+      createdDate = Instant.now(),
+      destroyedDate = None,
+      labels = Map.empty,
+      jupyterExtensionUri = Some("extension_uri"))
+
+    val c3 = Cluster(
+      clusterName = "name3",
+      googleId = UUID.randomUUID(),
+      googleProject = "dsp-leo-test",
+      googleServiceAccount = "not-a-service-acct@google.com",
+      googleBucket = "bucket3",
+      clusterUrl = Cluster.getClusterUrl("dsp-leo-test", "name3"),
+      operationName = "op3",
+      status = ClusterStatus.Deleted,
+      hostIp = None,
+      createdDate = Instant.now(),
+      destroyedDate = None,
+      labels = Map("foo" -> "bar"),
+      jupyterExtensionUri = Some("extension_uri"))
+
+    dbFutureValue { _.clusterQuery.save(c1) } shouldEqual c1
+    dbFutureValue { _.clusterQuery.save(c2) } shouldEqual c2
+    dbFutureValue { _.clusterQuery.save(c3) } shouldEqual c3
+
+    dbFutureValue { _.clusterQuery.listByLabels(Map.empty) }.toSet shouldEqual Set(c1, c2, c3)
+    dbFutureValue { _.clusterQuery.listByLabels(Map("bam" -> "yes")) }.toSet shouldEqual Set(c1)
+    dbFutureValue { _.clusterQuery.listByLabels(Map("bam" -> "no")) }.toSet shouldEqual Set.empty
+    dbFutureValue { _.clusterQuery.listByLabels(Map("bam" -> "yes", "vcf" -> "no")) }.toSet shouldEqual Set(c1)
```

Comment:
nit: can you add `bam -> yes` to another cluster that's not c1 so we can make sure that the intersection is working correctly

Reply:
👍